api/books/addToReadingList.ts

```diff
@@ -0,0 +1,22 @@
+import { baseWithAuth } from "api/base"
+
+export const addToReadingList = async (id: number) => {
+  return await baseWithAuth()
+    .request({
+      url: "/api/user_books",
+      method: "POST",
```

Comment:
what I'd recommend instead:

axios.post(base_url, data, sharedConfig)

where `sharedConfig` is coming from your base file:
```
const sharedConfig = {
  return {
   headers: <>
  |
}
```

and try using this pattern for other request types

Reply:
pls lmk if there is a better way to do this. Sa `wretch` kasi it is composable. 

Maybe the `base_url` can be part of the axios defaults?

Reply:
Hi. I've been thinking about this for the past hour pero I'm not sure na I completely understand how to implement your suggestion here. Sorry 😅

Reply:
Something like this

```
## api/base.ts
const authHeaders = () => {
  const token = localStorage.get()
  return {
    Authentication: token
  }
}

const baseWithAuth = () => {
  return axios.create({
    baseUrl: <environmnt variable>,
    headers: authHeaders()
  })
}

## api/user_books.ts
return await baseWithAuth()
    .request({
      url: "/api/user_books",
      method: "POST",
      data: {
        user_book: {
          book_id: id,
          start_date: new Date().toISOString(),
        },
      },
    })
```

Reply:
it works 🎉
![image](https://user-images.githubusercontent.com/91857581/189073140-63a9b05a-11e2-4864-b0f4-90a028e4e06e.png)

just wanted to make sure that I understand this. 
we took out the part where you get the token from localstorage, then put it in a function of its own.

then we called that function when setting the header of the axios instance.
is this for seperation of concerns? or am I missing something here? 🤔

Reply:
> is it better if I still make a baseWithoutAuth function?

no need if you already have another function like it.

> should I also include the

yes pls